contracts/modules/token/TokenController.sol

```diff
@@ -347,6 +352,35 @@ contract TokenController is ITokenController, LockHandler, LegacyMasterAware {
     token.transfer(user, totalAmount);
   }
 
+
+  function mintPooledStakingNXMRewards(uint amount, uint poolId) external {
+
+    require(msg.sender == address(cover), "TokenController: only Cover allowed");
+    mint(address(this), amount);
+    stakingPoolNXMBalances[poolId].rewards += amount.toUint96();
+  }
+
+  function burnPooledStakingNXMRewards(uint amount, uint poolId) external {
+
+    require(msg.sender == address(cover), "TokenController: only Cover allowed");
+    burnFrom(address(this), amount);
+    stakingPoolNXMBalances[poolId].rewards -= amount.toUint96();
+  }
+
+  function depositStakedNXM(address from, uint amount, uint poolId) external {
+    require(msg.sender == address(cover.stakingPool(poolId)), "TokenController: msg.sender not staking pool");
```

Comment:
We should have an internal library with a hardcoded proxy bytecode hash that will calculate the address of a staking pool given its id. having an external call for this is really unnecessary. See uniswap's library as an example: https://github.com/Uniswap/v2-periphery/blob/master/contracts/libraries/UniswapV2Library.sol#L17

Reply:
in theory it sounds good but this function needs to 2 parameters that are implicit there

`stakingPoolProxyCodeHash`

`address(cover)```` which is ```address(this)```

if we just make a library they need to be read anyway and passed in feels more awkward. the first one is from Cover.sol

Reply:
Let's implement this post V2